### ISSUE TITLE
iFrame allowed origins to be configurable and validated on the server

### ISSUE THREAD

**Issue by techsmyth** [cherrytwist/Server#4796]
## Description

As a sys admin I want a secure system
As a product owner I want the set of allowed origins for iFrames to be updated without code updates

## Acceptance criteria

- [ ] Configuration added to Alkemio server to specify the list of allowed origins for iFrames content
- [ ] COnfiguration picked up via api on client and used to check the origin URL within the browser
- [ ] Markdown scalar type on the server updated to check that all iFrame usage is from a whitelisted origin
Via JSON file with whitelist
- [ ] Additional domains to be whitelisted when working on this issue:
    - [ ] https://www.youtube-nocookie.com
    - [ ] https://open.spotify.com/
    - [ ] https://acrobatservices.adobe.com I think?
    - [ ] you can set your own domain: https://docs.h5mag.com/advanced-features/url-structure
- [ ] Inform Denise :) 

## Additional Context
This is both a security and maintainability issue. It is not sufficient to check the origin on the client, this must be enforced also on the server.

Example Spotify podcast by the one and only Neil Smyth:
Embedded link (edit issue to see the embedded code)
<iframe style="border-radius:12px" src="https://open.spotify.com/embed/episode/1zZkg8IK3KeIvcQlg6ER1B?utm_source=generator" width="100%" height="352" frameBorder="0" allowfullscreen="" allow="autoplay; clipboard-write; encrypted-media; fullscreen; picture-in-picture" loading="lazy"></iframe>
Episode link: https://open.spotify.com/episode/1zZkg8IK3KeIvcQlg6ER1B?si=2a8c430b266949dc

Example Adobe
Adobe documentation on embedding: https://developer.adobe.com/document-services/docs/overview/pdf-embed-api/gettingstarted/
Link: https://acrobatservices.adobe.com/view-sdk-demo/PDFs/Bodea%20Brochure.pdf

H5Mag
Link: https://magazine.gezondstedelijklevenhub.nl/dsih_magazine_wat_is_citizen_science/cover_citizenscience
iFrame: no example?

Youtube
Example 1: 
```
<iframe width="560" height="315" src="https://www.youtube.com/embed/uZo7-bqoELk?si=0xiK0W9AYn8vsnb0" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen></iframe>
```
or
Example 2: 
```
<iframe width="560" height="315" src="https://www.youtube-nocookie.com/embed/uZo7-bqoELk?si=0xiK0W9AYn8vsnb0" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen></iframe>
```

See the comments for details!

**Comment by techsmyth** [comment]
@SimoneZaza @bobbykolev from other issues I saw going by it looks like the iFrame allowed origins is enforced on the client; this should be a server controlled list of white listed origins

**Comment by bobbykolev** [comment]
Yes, this should have been extracted as it was missed in the initial implementation. Thanks for creating the ticket @techsmyth 
@reactoholic do not create duplicates, FYI

**Comment by bobbykolev** [comment]
Server work completed in: https://github.com/alkem-io/server/pull/4884

**Comment by bobbykolev** [comment]
Updates on the additional domains and how to use the whitelist.
@SimoneZaza , @Comoque1 

To update the list you should first obtain the current value.
```
query configuration {
  platform {
    settings {
      integration {
        iframeAllowedUrls
      }
    }
  }
}
```
Then copy the value, apply updates, and save the updates with the following:
```
mutation UpdateSettings ($settingsData: UpdatePlatformSettingsInput!) {
  updatePlatformSettings (settingsData: $settingsData) {
    integration {
      iframeAllowedUrls
    }
  }
}

# Variables
{
  "settingsData": {
    "integration": {
      "iframeAllowedUrls": [
        "https://issuu.com",
        "https://www.youtube.com",
        "https://player.vimeo.com",
        "https://www.canva.com",
        "https://demo.arcade.software",
        "https://open.spotify.com",
        "https://acrobatservices.adobe.com",
	"https://magazine.gezondstedelijklevenhub.nl",
        "https://www.youtube-nocookie.com"
      ]
    }
  }
}

```

IMPORTANT!
Some of the requested resources don't provide an iFrame code which the client requires.
So for documents and custom domains, the users should add the source into an iframe code e.g.
```
<iframe src="https://magazine.gezondstedelijklevenhub.nl/dsih_magazine_wat_is_citizen_science/cover_citizenscience"></iframe>
```
Last note, non-16/9 items are not looking perfect (see the spotify screenshot and the empty space below). However, for now I think we're ok and we can improve further in the future.

<img width="1035" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBK1NtQ0E9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--8283b6604bf2ecfc4d12ecf97b92e02b0e7d3ca3/image.png" alt="image.png" />

<img width="1034" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBK1dtQ0E9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--111bc3134dee2be83c0f0a1142f7a3cbf96a6f38/image.png" alt="image.png" />

<img width="1037" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBK2FtQ0E9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--d3c1d903b70fc89873641797dc2f7d72754cb7a4/image.png" alt="image.png" />

**Comment by bobbykolev** [comment]
@SimoneZaza, I suggest notifying the dev team once someone makes updates to the allowed list so we can keep the config up-to-date on all environments.

**Comment by SimoneZaza** [comment]
[like]  Simone Rietmeijer reacted to your message:
________________________________

**Comment by SimoneZaza** [comment]
@bobbykolev thanks for the detailed instructions, I've added them to the manuals/wiki in the Platform Content and Information channel of the Product Team in Teams.

**Comment by SimoneZaza** [comment]
@larssondenise

**Comment by Comoque1** [comment]
Tested OK on ACC env

Tested updating the list of configured iframes - OK
Tested importing iframes in multiple locations
 - templates - and using this template - OK
 - community guidance - its visualization on community tab - OK
 - callouts - OK
 - comment in a chat - OK

There is 1 issue - it is hard to put the iframe in concrete position on a MD - [bug](https://app.zenhub.com/workspaces/alkemio-development-5ecb98b262ebd9f4aec4194c/issues/gh/alkem-io/client-web/7589) is raised
